src/Symfony/Component/Amqp/Queue.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Symfony\Component\Amqp;
+
+use Symfony\Component\Amqp\RetryStrategy\RetryStrategyInterface;
+
+class Queue extends \AMQPQueue
+{
+    private $bindings;
+    private $retryStrategy;
+    private $retryStrategyQueuePattern;
+
+    /**
+     * Constructor.
+     *
+     * Special arguments:
+     *
+     * * routing_keys:
+     *   * If not set, $name will be used.
+     *   * If null, the bind will not use a routing key
+     *   * If false, the queue will not be bound
+     *   * Otherwise, the values string[] will be used
+     * * flags: if set, setFlags() will be called with its value
+     * * exchange: exchange to bind the queue to (default exchange is used if not set)
+     * * retry_strategy: A retry strategy instance to use (see RetryStrategyInterface)
+     * * retry_strategy_queue_pattern:
+     *   * The queue pattern to use for messages that needs to wait (default to %exchange%.%time%.wait)
+     *   * The pattern is expanded according to the %exchange% and %time% values.
+     * * bind_arguments: An array of bind arguments
+     */
+    public function __construct(\AMQPChannel $channel, $name, array $arguments = array(), $declare = true)
```

Comment:
This constructor has several if/else statements that could be changed into if only statements when the default value is defined before the if. It's not really a must, but it reduces the amount of possible code paths